### PR TITLE
Cleanup some unused sharing methods from the old sharing code

### PR DESCRIPTION
**Pull request by MorrisJobke**
Found in my stashes ;)

### CODE REVIEW

File: lib/private/Settings/Admin/Sharing.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function getForm() {
 			'allowResharing'                       => $this->config->getAppValue('core', 'shareapi_allow_resharing', 'yes'),
 			'allowShareDialogUserEnumeration'      => $this->config->getAppValue('core', 'shareapi_allow_share_dialog_user_enumeration', 'yes'),
 			'enforceLinkPassword'                  => Util::isPublicLinkPasswordRequired(),
-			'onlyShareWithGroupMembers'            => Share::shareWithGroupMembersOnly(),
+			'onlyShareWithGroupMembers'            => $this->config->getAppValue('core', 'shareapi_only_share_with_group_members', 'no') == 'yes',
```
Comment (rullzer) [inline lib/private/Settings/Admin/Sharing.php]: https://github.com/nextcloud/server/blob/b23032e4c569f7ba54197171cbb02ed8119b6811/lib/public/Share/IManager.php#L347

Comment (MorrisJobke) [reply lib/private/Settings/Admin/Sharing.php]: Done

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
I know unit tests are failing ... looking into it

**Review by ChristophWurst — APPROVED** [review]
LGTM

**Review by skjnldsv — APPROVED** [review]
:elephant:

**Comment by MorrisJobke** [comment]
CI failure is due to an unrelated trash bin problem.

**Pull request merged**